### ISSUE TITLE
[naming-convention] Allow selecting only const variables

### ISSUE THREAD

**Issue by nstepien** [typescript-eslint/typescript-eslint#1612]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": [1,
      { "selector": "variable", "format": ["camelCase", "PascalCase"] },
      // { "selector": "variable", "modifiers": "const", "format": ["camelCase", "PascalCase", "UPPER_CASE"] }
    ]
  }
}
```

```TS
const CONSTANT_NUMBER = 1;
```

**Expected Result**
I should be able to allow `UPPER_CASE` for `const`s only.

**Actual Result**
I can't narrow down the `variable` selector.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.2` |
| `@typescript-eslint/parser`        | `2.19.2` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.8.0` |
| `npm`                              | `6.13.7` |

**Comment by glen-84** [comment]
Should this issue be closed? (#2291)

**Comment by nstepien** [comment]
Looks like it works, thanks! 👍
